File: src/NodeParser/BooleanTypeNodeParser.ts
import ts from "typescript";
import type { Context } from "../NodeParser.js";
import type { SubNodeParser } from "../SubNodeParser.js";
import type { BaseType } from "../Type/BaseType.js";
import { BooleanType } from "../Type/BooleanType.js";

export class BooleanTypeNodeParser implements SubNodeParser {
    public supportsNode(node: ts.KeywordTypeNode): boolean {
        return node.kind === ts.SyntaxKind.BooleanKeyword;
    }
    public createType(node: ts.KeywordTypeNode, context: Context): BaseType {
        return new BooleanType();
    }
}

File: internal/parser/boolean_type_node_parser.go
package parser

import (
	"github.com/microsoft/typescript-go/shim/ast"

	"github.com/vega/ts-json-schema-generator/internal/types"
)

// BooleanTypeNodeParser handles the `boolean` keyword
// (src/NodeParser/BooleanTypeNodeParser.ts).
type BooleanTypeNodeParser struct{}

func NewBooleanTypeNodeParser() *BooleanTypeNodeParser {
	return &BooleanTypeNodeParser{}
}

func (p *BooleanTypeNodeParser) SupportsNode(node *ast.Node) bool {
	return node.Kind == ast.KindBooleanKeyword
}

func (p *BooleanTypeNodeParser) CreateType(_ *ast.Node, _ *Context, _ *types.ReferenceType) types.Type {
	return &types.BooleanType{}
}
